src/WebJobs.Script.WebHost/Management/LinuxInstanceManager.cs

```diff
@@ -91,19 +70,42 @@ public bool StartAssignment(HostAssignmentContext context)
                 _webHostEnvironment.DelayRequests();
 
                 // start the specialization process in the background
-                Task.Run(async () => await AssignAsync(context));
-
-                return true;
+                return Task.Run(async () =>
+                {
+                    await AssignAsync(context);
+                    return true;
```

Comment:
Could technically also use `return await AssignAsync(context)` which has a `Task<bool>` return type, but I think `return true` is closer to the previous "fire and forget" pattern - still evaluating the downstream effects of this change.

Reply:
Make the method itself `async`, and then make this change:

``` diff
- return Task.Run(async () =>
- {
-     await AssignAsync(context);
-     return true;
- }
+ await AssignAsync(context);
+ return true;
```

You will then need to remove all the `return Task.FromResult(x);`, instead just have `return x;`